webots_ros2_driver/webots_ros2_driver/webots_launcher.py

```diff
@@ -82,3 +124,57 @@ def __init__(self, output='screen', world=None, gui=True, mode='realtime', strea
             ],
             **kwargs
         )
+
+    def execute(self, context: LaunchContext):
+        # Check if the user wants to convert URDF files into robots
+        if self.__robots:
+            worldPath = self.__world.perform(context)
+            if not worldPath:
+                sys.exit('World file not specified (has to be specified with world=path/to/my/world.wbt')
+
+            worldCopy = worldPath[:-4] + '_with_URDF_robot.wbt'
+            shutil.copyfile(worldPath, worldCopy)
+            worldName=context.launch_configurations['world']
+            context.launch_configurations['world']=worldName[:-4] + '_with_URDF_robot.wbt'
+
+            for robot in self.__robots:
+                file_input = robot.get('urdf_location')
+                robot_name = robot.get('name')
+                robot_translation = robot.get('translation')
+                robot_rotation = robot.get('rotation')
+
+                if not file_input:
+                    sys.exit('URDF file not specified (has to be specified with \'urdf_location\': \'path/to/my/robotUrdf.urdf\'')
+                if not robot_name:
+                    sys.stderr.write('Robot name not specified (should be specified if more than one robot is present with \'name\': \'robotName\'\n')
+
+                convert2urdf(inFile=file_input, worldFile=worldCopy, robotName=robot_name, initTranslation=robot_translation, initRotation=robot_rotation)
+
+        return super().execute(context)
+
+    def getDriversList(self):
```

Comment:
We can still spawn controllers as we were doing it so far. Why do we need this?

Reply:
As the conversion and the `driver` node need the URDF file I wanted to ask the user to give it only once, but this required to specify more information in the `robotsList`.

Do you think it is better to stick as it is now? So there would be less differences between launching a `driver` node for an URDF / non-URDF robot ?

Reply:
I see. However, I don't think it is a big problem for a user to pass URDF two times. It is more consistent with the current behavior.